guides/source/security.md

```diff
@@ -453,7 +453,20 @@ Because of this, most web applications will display a generic error message "use
 
 However, what most web application designers neglect, are the forgot-password pages. These pages often admit that the entered username or e-mail address has (not) been found. This allows an attacker to compile a list of usernames and brute-force the accounts.
 
-In order to mitigate such attacks, _display a generic error message on forgot-password pages, too_. Moreover, you can _require to enter a CAPTCHA after a number of failed logins from a certain IP address_. Note, however, that this is not a bullet-proof solution against automatic programs, because these programs may change their IP address exactly as often. However, it raises the barrier of an attack.
+In order to mitigate such attacks, you can use rate limiting. Rails comes with a
```

Comment:
Rate limiting targets brute-force specifically, so maybe mention that?
```suggestion
In order to mitigate brute-force attacks, you can use rate limiting. Rails comes with a
```
Also, the NOTE above should probably mention rate-limiting as well?

Reply:
It also helps to mitigate credential stuffing, user name enumeration and similar attack vectors.